app/core/settings.py

```
@@ -251,5 +257,51 @@
+            raise ValueError(f"Rate limiting configuration is invalid: {e}")
+        
+        # Validate Redis connection URL structure
+        if settings.redis_url:
+            parsed = urlparse(settings.redis_url)
+            if parsed.scheme not in ("redis", "rediss") or not parsed.hostname:
+                raise ValueError(f"Invalid Redis URL format: {settings.redis_url}")
+        
+        # Validate model configuration service
+        try:
+            from app.core.model_config import get_model_config_service
+            service = get_model_config_service()
```

Ruff (0.12.2)

Local variable `service` is assigned to but never used

Remove assignment to unused variable `service`

(F841)